purview/utils/general.py

```diff
@@ -292,3 +291,76 @@ def get_node_count(query):
         return results_dict
 
     return results[0][0]
+
+
+def get_all_stories(query, resources_to_expand=None):
+    """
+    Query neo4j for all stories, filter them and serialize the results.
+
+    :param str query: raw cypher query
+    :param list resources_to_expand: List of resources to expand
+    :return results_dict: a dictionary containing serialized results received from neo4j
+    :rtype: dict
+    """
+    if resources_to_expand is None:
+        resources_to_expand = []
+
+    results_dict = {}
+    results, _ = db.cypher_query(query)
+
+    if not results:
+        return results_dict
+
+    # Creating a list of lists where each list is a collection of node ids
+    # of the nodes present in that particular story path.
+    # Paths are re-sorted in the ascending order for simplifying the logic below
+    path_nodes_id = []
+    for path in reversed(results):
+        path_nodes_id.append([node.id for node in path[0].nodes])
+
+    print("path nodes:", path_nodes_id)
+    unique_paths = []
+    for index, set(node_set) in enumerate(path_nodes_id[:-1]):
+        unique = True
+        for set(alternate_set) in path_nodes_id[index + 1:]:
+            # If the node_set is a subset of alternate_set,
+            # we know they are the same path except the alternate_set is longer.
+            # If alternate_set and node_set only have one node_id of difference,
+            # we know it's the same path but with the perspective of different siblings.
+            if node_set.issubset(alternate_set) or len(alternate_set.difference(node_set)) == 1:
+                unique = False
+                break
+        if unique:
+            # Since results is from longest to shortest, we need to get
+            # the opposite index.
+            unique_paths.append(results[(len(path_nodes) - index) - 1][0])
+    # While traversing, the outer for loop only goes until the second-last element
+    # because the inner for loop always starts one element ahead of the outer for loop.
+    # Hence, all the subsets of the last element will not be added to the unique_paths
+    # list as the for loops will eliminate them. So we add the last element
+    # since we are sure it is unique.
+    unique_paths.append(results[0][0])
+
+    unique_paths_nodes = []
+    for path in unique_paths:
+        unique_paths_nodes.append(list(path.nodes))
+
+    print(unique_paths)
+    for index, result in enumerate(unique_paths_nodes):
```

Comment:
As we talked about, reuse the code from `get_story` by breaking it out into a function. `results_dict` should really be a list of dicts.